internal/ebpfs/prog.go

```diff
@@ -69,3 +72,52 @@ func UnLoadMBProgs() error {
 	}
 	return nil
 }
+
+var (
+	ingress *ebpf.Program
+	egress  *ebpf.Program
+)
+
+func GetTCIngressProg() *ebpf.Program {
+	if ingress == nil {
+		err := initTCProgs()
+		if err != nil {
+			log.Errorf("init tc prog filed: %v", err)
+		}
+	}
+	return ingress
+}
+
+func GetTCEgressProg() *ebpf.Program {
+	if egress == nil {
+		err := initTCProgs()
+		if err != nil {
+			log.Errorf("init tc prog filed: %v", err)
+		}
+	}
+	return egress
+}
+
+func initTCProgs() error {
+	coll, err := ebpf.LoadCollectionSpec("bpf/mb_tc.o")
+	if err != nil {
+		return err
+	}
+	type progs struct {
+		Ingress *ebpf.Program `ebpf:"mb_tc_ingress"`
+		Egress  *ebpf.Program `ebpf:"mb_tc_egress"`
+	}
+	ps := progs{}
+	err = coll.LoadAndAssign(&ps, &ebpf.CollectionOptions{
```

Comment:
One question: how can we tell which section to load?

Reply:
This does not use `sec` to distinguish, it uses function names as separate procedures, see the description. `mb_tc_ingress` or `mb_tc_egress`.